o1vm/src/mips/witness.rs

```diff
@@ -1050,6 +1064,24 @@ impl<Fp: Field, PreImageOracle: PreImageOracleT> Env<Fp, PreImageOracle> {
         (opcode, instruction)
     }
 
+    /// Updates the instruction counter, accounting for the maximum number of
+    /// register and memory accesses per instruction.
+    /// Because MAX_NB_REG_ACC = 7 and MAX_NB_MEM_ACC = 12, at most the same
+    /// instruction will increase the instruction counter by MAX_ACC = 19.
+    ///
+    /// NOTE: actually, in practice it will be less than that, as there is no
+    ///       single instruction that performs all of them.
+    ///
+    /// This means that the actual number of instructions executed will result
+    /// from dividing the instruction counter by MAX_ACC (floor).
+    ///
+    /// Then, in order to update the instruction counter, we need to add 1 to
+    /// the real instruction counter and multiply it by MAX_ACC to have a unique
+    /// representation of each step (which is helpful for debugging).
+    pub fn update_instruction_counter(&mut self) {
```

Comment:
I think we might need to add another constraint. We might be able to add intermediate constraints. Will postpone the discussion to the soundness proof step, and consider this a good thing to move on.

Reply:
https://github.com/o1-labs/proof-systems/issues/2322